### PR TITLE
Add `to_hf_dataset` method to `olive.data.component.dataset.Dataset`

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
Add a method called `to_hf_dataset` which converts an olive Dataset to huggingface `datasets.Dataset`. This is useful if we want to use some huggingface related features such as `train_test_split` or easy indexing by feature/index. 

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/data/component/dataset.py [hunk 1]
```diff
@@ -60,6 +61,53 @@ def to_snpe_dataset(self):
...
+        """
+        from datasets import Dataset
+
+        # name for label column
+        label_cols = ["labels"]
+        if hasattr(self, "label_cols"):
+            label_cols = self.label_cols or ["labels"]  # some children classes may not have label_cols
+        label_name = label_name or label_cols[0]
+
+        if hasattr(self, "data") and isinstance(self.data, Dataset):
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/data/component/dataset.py:78]: It is not true for the dummydataset and rawdataset. they inherit the parent class but don't initialize super.

File: olive/data/component/dataset.py [hunk 2]
```diff
@@ -60,6 +61,53 @@ def to_snpe_dataset(self):
...
+                # label_cols is a list but we only use the first element for now
+                hf_dataset.remove_columns(col_name)
+            # rename the label column
+            if label_cols[0] != label_name:
+                if label_name in hf_dataset.column_names:
+                    raise ValueError(f"Cannot rename label column to {label_name} since it already exists")
+                hf_dataset = hf_dataset.rename_column(label_cols[0], label_name)
+            if self.max_samples is not None:
+                # truncate the dataset if max_samples is not None
+                hf_dataset = hf_dataset.select(range(len(self)))
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/data/component/dataset.py:92]: Looks like it will be fast since it will just do slicing for range 
image
Will update this.

[the remark this answers is not in the document]
Comment (jambayk) [reply olive/data/component/dataset.py]: Done!

File: olive/data/component/dataset.py [hunk 3]
```diff
@@ -60,6 +61,53 @@ def to_snpe_dataset(self):
...
+        This function is used to convert the dataset to huggingface dataset
+
+        :param label_name: The name of the label column in the new dataset
+        """
+        from datasets import Dataset
+
+        # name for label column
+        label_cols = ["labels"]
+        if hasattr(self, "label_cols"):
+            label_cols = self.label_cols or ["labels"]  # some children classes may not have label_cols
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/data/component/dataset.py]: Thanks for the catch! 
I have removed this part entirely. Now the new label col is the value of `label_name` parameter. That makes the logic and behavior simpler.

**Review by trajepl — APPROVED** [review]
